lib/Net/Twitter/Core.pm

```diff
@@ -179,16 +180,21 @@ sub _prepare_request {
         $msg = HTTP::Request->new($http_method, $uri);
     }
     elsif ( $http_method eq 'POST' ) {
-        # if any of the arguments are (array) refs, use form-data
-        $msg = (first { ref } values %natural_args)
-             ? POST($uri,
-                    Content_Type => 'form-data',
-                    Content      => [
-                        map { ref $_ ? $_ : encode_utf8 $_ } %natural_args,
-                    ],
-               )
-             : POST($uri, Content => $self->_query_string_for(\%natural_args))
-             ;
+        if( $content_type eq 'application/json' ) {
+            $msg = POST( $uri,  Content_Type => 'application/json', Content =>  encode_json $args );
```

Comment:
Use `encode_json \%natural_args` here, instead. That will prevent sending _synthetic args_, which are arguments used internally by Net::Twitter and aren't recognized by Twitter. Synthetic args start with a dash, e.g. `-authenticate`.